crates/bevy_animation/src/lib.rs

```diff
@@ -1,301 +1,30 @@
-    }
-
-    /// Unpause the animation
-    pub fn resume(&mut self) {
-        self.paused = false;
-    }
-
-    /// Is the animation paused
-    pub fn is_paused(&self) -> bool {
-        self.paused
-    }
-
-    /// Speed of the animation playback
-    pub fn speed(&self) -> f32 {
-        self.speed
-    }
-
-    /// Set the speed of the animation playback
-    pub fn set_speed(&mut self, speed: f32) -> &mut Self {
-        self.speed = speed;
-        self
-    }
-
-    /// Time elapsed playing the animation
-    pub fn elapsed(&self) -> f32 {
-        self.elapsed
-    }
-
-    /// Seek to a specific time in the animation
-    pub fn set_elapsed(&mut self, elapsed: f32) -> &mut Self {
-        self.elapsed = elapsed;
-        self
-    }
-}
-
-/// System that will play all animations, using any entity with a [`AnimationPlayer`]
-/// and a [`Handle<AnimationClip>`] as an animation root
-pub fn animation_player(
-    time: Res<Time>,
-    animations: Res<Assets<AnimationClip>>,
-    mut animation_players: Query<(Entity, &mut AnimationPlayer)>,
-    names: Query<&Name>,
-    mut transforms: Query<&mut Transform>,
-    children: Query<&Children>,
-) {
-    for (entity, mut player) in &mut animation_players {
-        if let Some(animation_clip) = animations.get(&player.animation_clip) {
-            // Continue if paused unless the `AnimationPlayer` was changed
-            // This allow the animation to still be updated if the player.elapsed field was manually updated in pause
-            if player.paused && !player.is_changed() {
-                continue;
-            }
-            if !player.paused {
-                player.elapsed += time.delta_seconds() * player.speed;
-            }
-            let mut elapsed = player.elapsed;
-            if player.repeat {
-                elapsed %= animation_clip.duration;
-            }
-            if elapsed < 0.0 {
-                elapsed += animation_clip.duration;
-            }
-            'entity: for (path, curves) in &animation_clip.curves {
-                // PERF: finding the target entity can be optimised
-                let mut current_entity = entity;
-                // Ignore the first name, it is the root node which we already have
-                for part in path.parts.iter().skip(1) {
-                    let mut found = false;
-                    if let Ok(children) = children.get(current_entity) {
-                        for child in children.deref() {
-                            if let Ok(name) = names.get(*child) {
-                                if name == part {
-                                    // Found a children with the right name, continue to the next part
-                                    current_entity = *child;
-                                    found = true;
-                                    break;
-                                }
-                            }
-                        }
-                    }
```

Comment:
I think I missed the 3D version review because this looks horrible. This is doing a children traversal and string lookup per animation player every single frame without any form of caching!?